backend/app/api/endpoints/adapter/shells.py

```
@@ -970,9 +991,10 @@ async def update_validation_status(
                 status_code=500, detail="Failed to update validation status"
             )
 
-        # Cleanup validation container if validation is completed
+        # Cleanup validation container if validation is completed.
+        # Run cleanup asynchronously to avoid blocking callback response.
         if request.executor_name and request.valid is True:
-            await _cleanup_validation_container(request.executor_name)
+            asyncio.create_task(_cleanup_validation_container(request.executor_name))
```

Ruff (0.15.7)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)